src/main/ruby/core/type.rb

```diff
@@ -255,6 +250,14 @@ def self.conversion_mismatch(val, cls, meth, res)
       raise TypeError, "can't convert #{val.class} to #{cls} (#{val.class}##{meth} gives #{res.class})"
     end
 
+    def self.fits_into_int?(val)
+      Integer === val && Truffle.invoke_primitive(:fixnum_fits_into_int, val)
```

Comment:
This primitive should probably be renamed.

Reply:
You mean `fixnum_fits_into_int`? Could do.

Reply:
Yes. No need to do it in this PR, but something to note.

Reply:
I'll merge this and do it in another PR then.

Reply:
#1352